velox/functions/lib/string/StringImpl.h

```diff
@@ -147,6 +147,42 @@ FOLLY_ALWAYS_INLINE int32_t charToCodePoint(const T& inputString) {
   return codePoint;
 }
 
+/// Returns the sequence of character Unicode code point of ASCII only input
+/// string
+template <typename T>
+FOLLY_ALWAYS_INLINE std::vector<int8_t> asciiStringToCodePoints(
```

Comment:
This function is not needed. It is also not necessary to copy the string. Just cast:

```
static_cast<const uint8_t*>(inputString.data())
```

Reply:
Do you mean for all ASCII case, do local static cast when code points are needed?

Reply:
Yes.

```
void callAscii(
      out_type<int64_t>& result,
      const arg_type<Varchar>& left,
      const arg_type<Varchar>& right) {
    auto* leftCodePoints =
        static_cast<const uint8_t*>(left.data());
    auto* rightCodePoints =
        static_cast<const uint8_t*>(right.data());
    doCall<int8_t>(result, leftCodePoints, rightCodePoints);
  }
```

Reply:
Got it, I was thinking it can stay in stringImpl as a util which can be used by other string transformation, currently will keep the casting in call method itself.